### DISCUSSION TITLE
post_233939

### DISCUSSION THREAD

**Discussion by giscus** [cincuranet/blog-comments#1361] [Comments]
# post_233939

https://www.tabsoverspaces.com/233939-writing-string-isnullorempty-using-pattern-matching-and-comparing-the-resulting-code?utm_source=newsletter.csharpdigest.net&utm_medium=newsletter&utm_campaign=net-9-linq-performance-edition

**Comment by JohanBonnetGVA** [comment]
Good to know, thank you! Now what about IsNullOrWhiteSpace overly used with Json and order input...

**Reply by cincuranet** [reply]
To check for white space you need to call `char.IsWhiteSpace` and there's really no straightforward way to do it with pattern matching.

**Comment by JeromeJ** [comment]
Presumably more of the same but I think this alternative would also achieve the same:

        public static int Proposal(string s)
        {
            if (s is { Length: > 0 })
                return s.Length * 2;

            return -1;
        }

By inverting the condition, you, presumably, do not need to check for the null anymore as it is implied in the pattern matching.
